db/scripts/fix_mismatched_induction_statuses.rb

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# This script finds teachers whose latest induction period has a pass/fail outcome
+# but their TRS status is still 'InProgress'. It then:
+# 1. Syncs with TRS to get latest status
+# 2. If still mismatched, sends the appropriate update to TRS
+
+require 'trs/api_client'
+
+# First get all teachers with mismatched statuses
+sql = <<-SQL
+  WITH latest_induction_period_start AS (
+    SELECT teacher_id, MAX(started_on) as latest_started_on
+    FROM induction_periods
+    GROUP BY teacher_id
+  ),
+  latest_induction_periods AS (
+    SELECT ip.*
+    FROM induction_periods ip
+    INNER JOIN latest_induction_period_start lips
+      ON lips.teacher_id = ip.teacher_id
+      AND lips.latest_started_on = ip.started_on
+  )
+  SELECT
+    t.id as teacher_id,
+    t.trn,
+    lip.outcome,
+    t.trs_induction_status
+  FROM teachers t
+  INNER JOIN latest_induction_periods lip ON t.id = lip.teacher_id
+  WHERE lip.outcome IN ('pass', 'fail')
+  AND t.trs_induction_status = 'InProgress'
+SQL
+
+mismatched_teachers = Teacher.find_by_sql(sql)
+
+Rails.logger.debug "Found #{mismatched_teachers.count} teachers with mismatched statuses"
```

Comment:
It feels a _little_ odd here that it's doing `Teacher.find_by_sql` but the SQL doesn't return a teacher row.

Maybe if the query above was amended so it just returned the `t.id as teacher_id`

```sql
  SELECT t.id as teacher_id
  FROM teachers t
  INNER JOIN latest_induction_periods ...
```

then we could pull the IDs out into an array:

```
mismatched_teacher_ids = ActiveRecord::Base.connection.execute(sql).map { |r| r["id"] }
```

Then we could pull them out cleanly:

```ruby
mismatched_teachers = Teacher.where(id: mismatched_teacher_ids)
```